dot_lpdev_cmds.sh

```diff
@@ -0,0 +1,719 @@
+    controllerAddress=$(cd $srcDir/protocol && truffle networks | grep Controller | cut -d':' -f2 | tr -cd '[:alnum:]')
+  fi
+
+}
+
+function __lpdev_protocol_reset {
+
+  if [ -d $srcDir/protocol/build ]
+  then
+    rm -rf $srcDir/protocol/build
+  fi
+
+  protocolBuilt=false
+  protocolBranch="repo not found"
+  unset controllerAddress
+
+}
+
+function __lpdev_protocol_init {
+
+  if [ -d $srcDir/protocol ];
+  then
+    echo "Protocol src directory exists"
+  else
+    echo "Cloning github.com/livepeer/protocol into src directory"
+    OPWD=$PWD
+    cd $srcDir
+    git clone -b develop "https://github.com/livepeer/protocol.git"
+    cd $OPWD
+  fi
+
+  ##
+  # Install local dev truffle.js
+  ##
+
+  __lpdev_geth_refresh_status
+  if [ -z "${gethMiningAccountngAccount}" ]
+  then
+    echo "Geth Mining Account not found"
+    return 1
+  fi
+
+  if grep -q ${gethMiningAccount:-"none"} $srcDir/protocol/truffle.js
+  then
+    echo "Local dev version of $srcDir/protocol/truffle.js already exists"
+  else
+    echo "Installing local dev version of $srcDir/protocol/truffle.js"
+
+    cat << EOF > $srcDir/protocol/truffle.js
+module.exports = {
+    networks: {
+        development: {
+            host: "localhost",
+            port: 8545,
+            network_id: "*" // Match any network id
+        },
+        lpTestNet: {
+            from: "0x$gethMiningAccount",
+            host: "localhost",
+            port: 8545,
+            network_id: 54321,
+            gas: 6700000
+        }
+    }
+};
+EOF
+
+    echo "Installing local dev version of $srcDir/protocol/migrations/migrations.config.js"
+
+    cat << EOF > $srcDir/protocol/migrations/migrations.config.js
+module.exports = {
+    bondingManager: {
+        numActiveTranscoders: 5,
+        unbondingPeriod: 2
+    },
+    jobsManager: {
+        verificationRate: 10,
+        jobEndingPeriod: 50,
+        verificationPeriod: 50,
+        slashingPeriod: 50,
+        failedVerificationSlashAmount: 20,
+        missedVerificationSlashAmount: 30,
+        finderFee: 4
+    },
+    roundsManager: {
+        blockTime: 1,
+        roundLength: 5
+    },
+    faucet: {
+        faucetAmount: 100000000000000000000,
+        requestAmount: 1000000,
+        requestWait: 2,
+        whitelist: []
+    },
+    minter: {
+        initialTokenSupply: 10000000 * Math.pow(10, 18),
+        yearlyInflation: 26
+    },
+    verifier: {
+        verificationCodeHash: "QmWdbVR8SUS9TU5a9HFP2qG18ck6Vh4mL2PYsHcB9sHXN7",
+        solvers: ["0x0ddb225031ccb58ff42866f82d907f7766899014"],
+        gasPrice: 20000000000,
+        gasLimit: 3000000
+    }
+}
+EOF
+
+  fi
+
+  ##
+  # Update npm
+  ##
+
+  if [ -d $srcDir/protocol/node_modules ]
+  then
+    echo "Npm packages already installed"
+  else
+    echo "Running \`npm install\`"
+    OPWD=$PWD
+    cd $srcDir/protocol
+    npm install
+    cd $OPWD
+  fi
+}
+
+function __lpdev_protocol_deploy {
```

Comment:
Might need to include a check somewhere here to make sure the protocol init stuff like npm install didn't fail. I was running into an error when deploying due to the node modules not being installed

Reply:
As @dob has discovered in his feedback, the npm part is going to be the weirdest. I'm going to see if there's a quick sane way to create a uniq node environment that doesn't depend on `protocol/node_modules`